### DISCUSSION TITLE
How do you manage default locale in Multilanguage admins?

### DISCUSSION THREAD

**Discussion by adriallongarriu** [Laravel-Backpack/community-forum#1144] [Q&A (Help)]
I have a website with 3 locales. One a new entry is created in the admin for example an Article it is created in the current locale and after i can edit and create the content in other locales and the fields like title or content, the translatable fields, are preload with the content of first locale.

But if i use `Laravel-Backpack/language-switcher`  the user can select a locale which is not the default. For this reason when creating the subsequent entries of the same article the fields are blank puls in the front the content is only in one locale and the rest does not show anything. For this reason, the content is not displayed consistently as it varies depending on the language used by the user.

Ho do you manage this case?
- Review your queries so that in the frontend it only shows content in the language used by the web. For example if article do not have title in EN and the web is using ES you dont show this article?
- Forces articles to always be created in fallback_locale first? if you wanted to make this option how would you do it in backpack?
- You have other options?

**Comment by tabacitu** [comment]
Hi @adriallongarriu ,

This is why I always try to delay adding multi-language to a project. Clients think "_oh it'll be easy, we'll have everything translated_" but the reality is... they won't. They will have some things translated, some not.

I understand the need to only show the things that are translated. And you CAN do that, if you want, by using custom queries in your app. I wouldn't do that. I would just show them in the default language, when they don't have that translation. It's shitty... but... at least it preserves data integrity.

I don't see this as a Backpack problem per-se... it's an admin problem... they aren't dilligent enough to translate everything. But if you want to make it easier for them to spot which entities are NOT completely translated, perhaps you can create a custom column... for `translations`, that shows something like: `🇪🇸🇬🇧❌❌❌` - so that it's obvious if there's an X, they have more translating to do?

**Reply by adriallongarriu** [reply]
I would also prefer to show it in the default language so I don't have to change the queries but the problems is backpack when using multi language in the admin the first entry created is not the default/fallback locale, is created in the current locale. This is what cause more inconsistencies and cause latery to show nothing in some cases becouse there is no content in the fallback locale. It is random depending on the user.

For example i have 3 locales ES, EN, CA. The fallback locale is ES. 

If i create a new article and fill the title. An the user use `Laravel-Backpack/language-switcher` change to other language the entry is created in one locale and may or may not be the default.
| Selected locale  | what is stored in deb | Has default locale | 
| ------------- | ------------- | ------------- |
| EN  | { en: "Hello"}  |  |
| ES  | { es: "Hola"}  |  X |
| CA  | { ca: "Hola"}  | |

If the user use EN the default locale is empry. You can test it in the demo.

**Comment by pxpm** [comment]
Hey @adriallongarriu what version of backpack are you using ? I think recently (v6.7.x) we introduced the ability for you to add `public $useFallbackLocale = true;` in your Model, and the behavior should be:
- show selected locale if translated
- show fallback locale if translated
- show first translation found

Let me know if that helps. 

Cheers

**Reply by adriallongarriu** [reply]
Hey @pxpm. I alredy have (v6.7) and updated to last minor versión 6.7.38 and add `public $useFallbackLocale = true;` to the model but a don't see any chages. Is there somewhere in the documentation that explains more about how it works?

I create an new entry in `CA` in database i have the field **NAME** `{"ca":"Nom ca"}` but when i change the ui to `ES` o other language the translation is empty and dont show the name only '-'. 
- show selected locale if translated -> entry is no translated in the current locale(`ES`)
- show fallback locale if translated -> fallback locale is `ES` is not translated
- show first translation found -> This part is no wroking and is not showing any
![image](https://github.com/user-attachments/assets/4123f070-6b1a-40c7-b6de-9ac381321a9a)

**Reply by pxpm** [reply]
Hey @adriallongarriu it's possible that you have overwritten some blade files and you didn't get the changes ? 

Here is the PR that introduced that change: https://github.com/Laravel-Backpack/CRUD/pull/4267

Let me know how it goes. 👍 If the error persist I will investigate it further, but should be working as shown in that PR.

**Reply by adriallongarriu** [reply]
I overwritten some blade for fields but i haven't touch any for the list views or text column.
But reading yout PR maybe what i need to solve my problems is the option `fallbackAny` from spatie laravel-translatable
https://spatie.be/docs/laravel-translatable/v6/basic-usage/handling-missing-translations#content-falling-back-to-any-locale because i need to show the text in the admin/backpack but the aricles are also shown on the frontend.
